src/recommend.rs

```diff
@@ -0,0 +1,110 @@
+use std::ops::Index;
+
+use hyper::{Body, Client, Uri};
+use serde_derive::{Deserialize, Serialize};
+
+type ConnectivityResponse = Vec<Nodes>;
+
+#[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct Nodes {
+    pub public_key: String,
+    pub channels: i64,
+    pub capacity: i64,
+}
+
+use serde_json::Value;
+
+#[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Node {
+    #[serde(rename = "public_key")]
+    pub public_key: String,
+    pub alias: String,
+    #[serde(rename = "first_seen")]
+    pub first_seen: i64,
+    #[serde(rename = "updated_at")]
+    pub updated_at: i64,
+    pub color: String,
+    pub sockets: String,
+    #[serde(rename = "as_number")]
+    pub as_number: i64,
+    #[serde(rename = "city_id")]
+    pub city_id: Value,
```

Comment:
Sometimes these fields are `null`. Can we skip deserializing the ones we don't need? Or use Option

```
{"public_key":"028624b5cb872511d8b23a25ec351f6201af7e6b640edc0888fce05a8dede2bc8a","alias":"[SatSale]|LabNode","first_seen":1653621833,"updated_at":1666958783,"color":"#50beef","sockets":"24nrvuglk652fenrjorggbi4d53rly56w7xhetux3fzf6wbbxdahi4yd.onion:8888","as_number":null,"city_id":null,"country_id":null,"subdivision_id":null,"longitude":null,"latitude":null,"iso_code":null,"as_organization":null,"city":null,"country":null,"subdivision":null,"active_channel_count":1,"capacity":"1000000","opened_channel_count":1,"closed_channel_count":2}
```

Reply:
e.g. like in https://github.com/chaincase-app/nolooking/pull/61/commits/0c0e97d597e74ee45c10828459a671e75af70b16

Reply:
Yes, unused fields should be excluded from our struct. serde will ignore them. No need for Options fields except to represent truly optional values.

Also. What's going on with the renames?

Reply:
Renames are changing variable name casing.

Reply:
camel case to snake case

Reply:
fixed in [a897962](https://github.com/chaincase-app/nolooking/pull/51/commits/a8979622e434c1c396f1aa5f60628f1fa8658e53)